cmd/pctl/main.go

```diff
@@ -93,6 +93,21 @@ func parseArgs(c *cli.Context) (string, *client.Client, error) {
 	return c.Args().First(), client, nil
 }
 
+func getClient(c *cli.Context) (*client.Client, error) {
+	options := client.ServiceOptions{
+		KubeconfigPath: c.String("kubeconfig"),
+		Namespace:      c.String("catalog-service-namespace"),
+		ServiceName:    c.String("catalog-service-name"),
+		ServicePort:    c.String("catalog-service-port"),
+	}
+
+	if c.Args().Len() > 0 {
+		return nil, fmt.Errorf("argument must not be provided")
+	}
+
+	return client.NewFromOptions(options)
+}
```

Comment:
is this a repeat of the above? It could be re-used instead of duplicated :smile:

Reply:
yeah I was thinking of refactoring it but got sidetracked by the failing build 😓  I'll refactor it 👍🏻

Reply:
done 👍🏻